application/gui/properties_window.py

```diff
@@ -193,6 +197,8 @@ def _rename_item(self, widget=None, data=None):
 
 	def _create_monitor(self):
 		"""Create item monitor"""
+		if hasattr(self, '_monitor') and self._monitor is not None and not self._monitor.is_cancelled():
+			self._monitor.cancel()
```

Comment:
I noticed that properties_window.py: _rename_item() would just create a new
monitor if the file was renamed and didn't see the previous monitor being
"freed". I added the two lines as a safety measure. Do you reckon they are
not needed (ever)?

PS: you have quite a pull request back log :-)

On Mon, May 2, 2016 at 10:15 AM, Mladen Mijatov <notifications@github.com>
wrote:

> In application/gui/properties_window.py
> <https://github.com/MeanEYE/Sunflower/pull/230#discussion_r61718547>:
>
> > @@ -193,6 +197,8 @@ def _rename_item(self, widget=None, data=None):
> >
> >  	def _create_monitor(self):
> >  		"""Create item monitor"""
> > +		if hasattr(self, '_monitor') and self._monitor is not None and not self._monitor.is_cancelled():
> > +			self._monitor.cancel()
>
> Why is this being done like this? Monitors should be automatically
> canceled on path cancel.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/MeanEYE/Sunflower/pull/230/files/dc4966501ec1bcc599007680f92fcff3e7067830#r61718547>
>